File: PHP/classes/Perro.php
<?php

namespace Veterinaria;

use Mascota;

class Perro extends Mascota {

    private $tamanho;

    public function __construct($tamanho, $nombre, $edad) {
        parent::__contruct($nombre, $edad);
        $this->tamanho = $tamanho;
    }

    public function perseguirLaCola() {
        return "Me persigo la cola";
    }

    public function comunicarse() {
        return "guau, guau";
    }

    public function presentarse() {
        return parent::presentarse() . " perro";
    }
}

File: Java/Veterinaria/src/veterinaria/Perro.java
package veterinaria;

public class Perro extends Mascota {
    private String tamanho;

    public Perro(String tamanho, String nombre, int edad) {
        super(nombre, edad);
        this.tamanho = tamanho;
    }
    
    public String perseguirLaCola() {
        return "Me persigo la cola";
    }

    @Override // Flag dice que el metodo sobreescribe el metodo padre
    public String comunicarse() {
        return "guau, guau";
    }
    
    @Override // Sobre escritura del padre para agregar
    public String presentarse() {
        return super.presentarse() + " perro";
    }
}
